internal/server/controllers/rateController.go

```diff
@@ -0,0 +1,37 @@
+package controllers
+
+import (
+	"errors"
+	"github.com/gin-gonic/gin"
+	"gses4_project/internal/apperrors"
+	"gses4_project/internal/services"
+	"net/http"
+)
+
+type IRateService interface {
+	Get() (*float64, error)
+}
+
+type RateController struct {
+	RateService IRateService
+}
+
+func NewRateController() *RateController {
+	return &RateController{
+		RateService: services.NewRateService(),
+	}
+}
+
+func (c *RateController) Get(ctx *gin.Context) {
+	rate, err := c.RateService.Get()
+	if err != nil {
+		var httpErr *apperrors.HttpError
+		if errors.As(err, &httpErr) {
```

Comment:
This is where the controller checks what errors occurred in the application and returns an http error. Usually these are 500 errors, and the error text is passed to the application `err.Error()`. Or we don't pass the error text, but log it if we don't want the client to know what kind of error we have. Errors 400-499 are returned when the request is validated (error validation of the request body, authorization validation, etc.)